src/scope/models/version.js

```diff
@@ -268,7 +268,8 @@ export default class Version extends BitObject {
         packageDependencies: this.packageDependencies,
         devPackageDependencies: this.devPackageDependencies,
         peerPackageDependencies: this.peerPackageDependencies,
-        envsPackageDependencies: this.envsPackageDependencies
+        // Do not store it if it's empty for backward compatibility
```

Comment:
I suggest being compatible with other packages (devPackageDependencies, peerPackageDependencies).
The packages are stored as `{}` when they are empty.
However, when calculating the hash, which is the id() function, the packages are removed.
As a result, you get the backward compatibility, as the hash is the same.

The last part is done here:
```
 const filterFunction = (val, key) => {
      if (key === 'devDependencies' || key === 'devPackageDependencies' || key === 'peerPackageDependencies') {
        return !R.isEmpty(val);
      }
```
So, all you need to do is adding 'envsPackageDependencies' to the `if` statement.